pkg/models/terminal/terminal.go

```diff
@@ -224,3 +349,60 @@ func (t *terminaler) HandleSession(shell, namespace, podName, containerName stri
 
 	session.Close(1, "Process exited")
 }
+
+func (t *terminaler) HandleShellAccessToNode(nodename string, conn *websocket.Conn) {
+
```

Comment:
https://github.com/kubesphere/kubesphere/blob/f0210193c1e9d986690e3bd8835ca444e401688d/pkg/kapis/terminal/v1alpha2/handler.go#L61-L82

Check permission to create pod/exec is required. Use the WebSocket protocol, only the GET verb will be received before the upgrade.

It's similar to `pods/exec`, `kubectl` using `SPDY/3.1` protocol connect to the pod, we grant pod exec permission by RBAC rule: `create pods/exec`

Reply:
I checked the permissions in ```terminalHandler.handleShellAccessToNode()```, same as you quoted. if the current user doesn't have correct permission, he can't enter this function.